### PR TITLE
AdOcean Adpater for Prebid 1.0

### PR DESCRIPTION
**Pull request by jdrobiecki**
## Type of change
- [x ] New bidder adapter
   

## Description of change

new bidder adapter .

- test parameters for validating bids
```
{
  bidder: 'adocean',
  params: {
    slaveId: 'adoceanmyaozpniqismex',
    masterId: 'tmYF.DMl7ZBq.Nqt2Bq4FutQTJfTpxCOmtNPZoQUDcL.G7',
    emiter: 'myao.adocean.pl'
  }
}
```

- contact email of the adapter’s maintainer: prebid@gemius.com
- [x ] official adapter submission

### CODE REVIEW

File: modules/adoceanBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    adCode += decodeURIComponent(placementResponse.code);
+
+    const bid = {
+      ad: adCode,
+      bidderCode: BIDDER_CODE,
+      cpm: parseFloat(placementResponse.price),
+      currency: placementResponse.currency,
+      height: parseInt(placementResponse.height, 10),
+      requestId: bidRequest.bidIdMap[placementResponse.id],
+      width: parseInt(placementResponse.width, 10),
```
Comment (matthewlane) [inline modules/adoceanBidAdapter.js]: `creativeId`, `netRevenue`, and `ttl` should also be in this object as well. See the table in http://prebid.org/dev-docs/bidder-adapter-1.html#interpreting-the-response for information about these fields

File: modules/adoceanBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  if (!placementResponse.error) {
+    let adCode = '<script type="application/javascript">(function(){var wu="' + (placementResponse.winUrl || '') + '",su="' + (placementResponse.statsUrl || '') + '".replace(/\\[TIMESTAMP\\]/,(new Date()).getTime());';
+    adCode += 'if(navigator.sendBeacon){if(wu){navigator.sendBeacon(wu)||((new Image(1,1)).src=wu)};if(su){navigator.sendBeacon(su)||((new Image(1,1)).src=su)}}';
+    adCode += 'else{if(wu){(new Image(1,1)).src=wu;}if(su){(new Image(1,1)).src=su;}}';
+    adCode += '})();<\/script>';
+    adCode += decodeURIComponent(placementResponse.code);
+
+    const bid = {
+      ad: adCode,
+      bidderCode: BIDDER_CODE,
```
Comment (matthewlane) [inline modules/adoceanBidAdapter.js]: `bidderCode` can be dropped as well, it'll be set automatically by `bidderFactory`. Docs are being updated now, sorry for the confusion https://github.com/prebid/prebid.github.io/pull/411

File: modules/adoceanBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      assignToMaster(v, bidRequestsByMaster);
+    });
+    requests = utils._map(bidRequestsByMaster, function(v, k) {
+      return buildRequest(v, k);
+    });
+
+    return requests;
+  },
+
+  interpretResponse: function(serverResponse, bidRequest) {
```
Comment (matthewlane) [inline modules/adoceanBidAdapter.js]: One more change with #1748 now merged to master:

The first argument to `interpretResponse` has been changed, and now looks like this:

```JavaScript
{
  body: responseBody,
  headers: {
    get: function(header) { /* returns a header from the HTTP response */ }
  }
}
```

So to retain the same functionality as before, doing something like
```JavaScript
serverResponse = serverResponse.body;
```
just below this line (or however you'd prefer) , and in tests as needed, should help

Could you rebase off of master and update that?

**Comment by jdrobiecki** [comment]
Fixed and rebased to master.

**Pull request merged**